### ISSUE TITLE
External application overlay

### ISSUE THREAD

**Issue by trigg** [Plagman/gamescope#288]
I am interested in embedding extra overlay information into gamescope. I plan to follow this shortly with a PR but would like to gather opinions and alterations to this feature request.

Ideally:
1) Run gamescope
2) External application opens an X window, adds atom 'STEAM_EXTERNAL_OVERLAY'
3) Gamescope picks up this window and composites it above the current window

This feature needs to be agnostic in usage so it would be possible for most applications to draw their own overlay. 

Potential use cases:
- allowing streaming services to do picture-in-picture while you play games
- clan text-chat
- voice channels status & online users

**Comment by Plagman** [comment]
Thanks! Beyond the MR probably needing some rebasing, having the Atom be called GAMESCOPE_ something probably makes more sense. It'd be cool to be able to specify a corner to align it to, maybe.

**Comment by trigg** [comment]
I will do that properly after work. I completely agree that a more generic naming convention would probably be better but it hadn't occurred to me at the time.

Currently the image of the overlay is stretched across the whole display much the same as the game so corner is decided by the application drawing the overlay, but I can see why it would be nice to control it from the compositor.

**Comment by trigg** [comment]
I've just done a test run against my own overlay and the PR is back to working order.

**Comment by nosmirck** [comment]
This is the only place where I can ask this question, I don't want to open a new unnecessary issue, so here it goes:

I'd like to know how can I add an overlay on top of a running game with gamescope.

My use case is similar to the OP. In my case, I'd like to create a decky loader plugin that renders a crosshair or reticle in the middle of the screen. Some games have a very small reticle (just one little circle or even a pixel) that's not really visible in the Deck's screen and this would allow players to add a custom crosshair that's more visible.

I don't even know where to start, I've successfully managed to do some kind of "hello world" plugin that runs console commands, that's it, I use it to run console commands from game mode. This means I can add some kind of application that runs in the background (launched from a console command) that uses this feature from this request (which apparently was added).

Is there some documentation on what we can do to create such overlay? Is there anything specific that needs to be done? Any examples?

I'm a mobile and web developer, I've never worked with low level features like these and I'd love to learn! If anyone can point me in the right direction I'd be grateful!

Thanks in advance!
